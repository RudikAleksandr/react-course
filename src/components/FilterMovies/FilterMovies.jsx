import React, { useCallback } from 'react';
import { Row, Col, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';
import cn from './FilterMovies.module.css';

const movieTypes = ['all', 'documentary', 'comedy', 'horror', 'crime'];

const FilterMovies = ({ activeType, onClickMovieGenre }) => {
  const handleClickMovieType = useCallback(({ target: { textContent } }) => {
    onClickMovieGenre(textContent);
  }, [onClickMovieGenre]);

  const movieTypeButtons = movieTypes.map((movieType, index) => (
    <Button
      key={index}
      variant=""
      onClick={handleClickMovieType}
      className={`${cn.filterButton} ${activeType === movieType ? cn.activeFilterButton : ''}`}
    >
      {movieType}
    </Button>
  ));

  return (
    <Row className={`${cn.filterMovies}`}>
      <Col className="pl-0">
        {movieTypeButtons}
      </Col>
    </Row>
  );
}

FilterMovies.propTypes = {
  activeType: PropTypes.string,
  onClickMovieGenre: PropTypes.func.isRequired
};

export default FilterMovies;
