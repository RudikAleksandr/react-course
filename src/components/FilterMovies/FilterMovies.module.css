
.filterMovies {}

.filterMovies .filterButton {
  float: left;
  color: white;
  font-size: 1.15rem;
  margin-right: 30px;
  padding: 9px 0 7px 0;
  text-transform: uppercase;
}

.filterMovies .filterButton:hover {
  color: white;
}

.filterMovies .filterButton:focus,
.filterMovies .filterButton:active {
  box-shadow: none;
}

.filterMovies .activeFilterButton::after {
  content: "";
  position: relative;
  top: 12px;
  display: block;
  width: 100%;
  height: 4px;
  background-color: #f65261;
}
