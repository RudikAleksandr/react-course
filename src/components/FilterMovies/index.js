import FilterMovies from './FilterMovies.jsx';

export default FilterMovies;
