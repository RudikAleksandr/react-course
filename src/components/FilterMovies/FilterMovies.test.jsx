import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import FilterMovies from './FilterMovies';

describe('FilterMovies component', () => {
  test('render FilterMovies snapshot', () => {
    const component = <FilterMovies activeType="all" onClickMovieGenre={() => { }} />;
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });

  test('should change genre type', () => {
    const mockClickMovieGenre = jest.fn();
    const movieGenre = 'documentary';
    const component = <FilterMovies onClickMovieGenre={mockClickMovieGenre} />;

    render(component);

    userEvent.click(screen.getByText(movieGenre), movieGenre);
    expect(mockClickMovieGenre).toHaveBeenCalledWith(movieGenre);
  });
});
