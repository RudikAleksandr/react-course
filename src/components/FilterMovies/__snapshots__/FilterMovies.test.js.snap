// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMovies component render FilterMovies snapshot 1`] = `
<DocumentFragment>
  <div
    class="filterMovies row"
  >
    <div
      class="pl-0 col"
    >
      <button
        class="filterButton activeFilterButton btn"
        type="button"
      >
        all
      </button>
      <button
        class="filterButton  btn"
        type="button"
      >
        documentary
      </button>
      <button
        class="filterButton  btn"
        type="button"
      >
        comedy
      </button>
      <button
        class="filterButton  btn"
        type="button"
      >
        horror
      </button>
      <button
        class="filterButton  btn"
        type="button"
      >
        crime
      </button>
    </div>
  </div>
</DocumentFragment>
`;
